steward/src/cellars/cellar_v2_5.rs

```diff
@@ -0,0 +1,355 @@
+
+            let call = AddPositionCall {
+                index: params.index,
+                position_id: params.position_id,
+                configuration_data: params.configuration_data.into(),
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2_5Calls::AddPosition(call).encode())
+        }
+        Function::CallOnAdaptor(params) => {
+            for adaptor_call in params.data.clone() {
+                check_blocked_adaptor(&adaptor_call.adaptor)?;
+            }
+
+            log_cellar_call(CELLAR_NAME, &CallOnAdaptorCall::function_name(), &cellar_id);
+            let call = CallOnAdaptorCall {
+                data: get_encoded_adaptor_calls(params.data)?,
+            };
+
+            Ok(CellarV2_5Calls::CallOnAdaptor(call).encode())
+        }
+        Function::RemovePosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemovePositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemovePositionCall {
+                index: params.index,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2_5Calls::RemovePosition(call).encode())
+        }
+        Function::SetHoldingPosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetHoldingPositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetHoldingPositionCall {
+                position_id: params.position_id,
+            };
+
+            Ok(CellarV2_5Calls::SetHoldingPosition(call).encode())
+        }
+        Function::SetStrategistPayoutAddress(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetStrategistPayoutAddressCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetStrategistPayoutAddressCall {
+                payout: sp_call_parse_address(params.payout)?,
+            };
+
+            Ok(CellarV2_5Calls::SetStrategistPayoutAddress(call).encode())
+        }
+        Function::SwapPositions(params) => {
+            log_cellar_call(CELLAR_NAME, &SwapPositionsCall::function_name(), &cellar_id);
+            let call = SwapPositionsCall {
+                index_1: params.index_1,
+                index_2: params.index_2,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2_5Calls::SwapPositions(call).encode())
+        }
+        Function::SetShareLockPeriod(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetShareLockPeriodCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetShareLockPeriodCall {
+                new_lock: string_to_u256(params.new_lock)?,
+            };
+
+            Ok(CellarWithShareLockPeriodV1Calls::SetShareLockPeriod(call).encode())
```

Comment:
Just to confirm, this existing function moved from the base cellar to the additional ABI we've added, which is one of the new permutations?

Reply:
Yes
https://github.com/PeggyJV/cellar-contracts/blob/main/src/base/permutations/CellarWithShareLockPeriod.sol